packages/theme-predictive-search-component/__tests__/theme-predictive-search-component.test.js

```diff
@@ -0,0 +1,391 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint-disable no-unused-vars */
+
+/**
+ * TODO
+ * - Add test for kill()
+ * - Add test for body event to close predictive search
+ */
+
+import xhrMock from "xhr-mock";
+import PredictiveSearchUI from "../src/theme-predictive-search-component";
+import searchAsYouTypeTheCallingFixture from "../__fixtures__/search_as_you_type_the_calling.json";
+
+const defaultTargets = [
+  {
+    input: "foo",
+    submit: "foo",
+    result: "foo"
+  }
+];
+
+function defaultResultTemplateFct(data) {
+  return `
+    <ul>
+      ${data.products.map(
+        item => `
+          <li>
+            <a href="${item.url}">
+              <img src="${item.title}" />
+              <span>${item.title}</span>
+              <span>${item.price}</span>
+            </a>
+          </li>
+        `
+      )}
+    </ul>
+  `;
+}
+
+beforeEach(() => {
+  jest.useFakeTimers();
+  xhrMock.setup();
+  document.body.innerHTML = `
+    <input id="input-default" type="text" />
+    <div data-predictive-search-mount="default" />
+    <input id="input-header" type="text" />
+    <div data-predictive-search-mount="header" />
+  `;
+});
+
+afterEach(() => {
+  xhrMock.teardown();
+});
+
+it("create an instance of PredictiveSearchUI", () => {
+  const predictiveSearchUI = new PredictiveSearchUI({
+    targetsSelectors: defaultTargets,
+    resultTemplateFct: defaultResultTemplateFct
+  });
+
+  expect(predictiveSearchUI).toBeInstanceOf(PredictiveSearchUI);
+});
+
+it("on input focus, trigger the callback onInputFocus()", () => {
+  const spy = jest.fn();
+  const predictiveSearchUI = new PredictiveSearchUI({
+    targetsSelectors: [
+      {
+        input: "#input-default",
+        submit: null,
+        result: '[data-predictive-search-mount="default"]'
+      }
+    ],
+    resultTemplateFct: defaultResultTemplateFct,
+    onInputFocus: spy
+  });
+
+  const $input = document.querySelector("#input-default");
+  const evt = new Event("focus");
+  $input.dispatchEvent(evt);
+
+  expect(spy).toBeCalledTimes(1);
+});
+
+it("on input focus, when the input has a value, trigger open()", () => {
+  const spy = jest.spyOn(PredictiveSearchUI.prototype, "open");
+  const predictiveSearchUI = new PredictiveSearchUI({
+    targetsSelectors: [
+      {
+        input: "#input-default",
+        submit: null,
+        result: '[data-predictive-search-mount="default"]'
+      }
+    ],
+    resultTemplateFct: defaultResultTemplateFct
+  });
+
+  xhrMock.get(/^\/search\/suggest\.json/g, (req, res) => {
```

Comment:
Ideally I would mock `PredictiveSearch` so we wouldn't have to mock the actual request.